pallets/bridge/src/lib.rs

```diff
@@ -0,0 +1,621 @@
+		}
+		r_id
+	}
+
+	#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+	pub enum ProposalStatus {
+		Initiated,
+		Approved,
+		Rejected,
+	}
+
+	#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+	pub struct ProposalVotes<AccountId, BlockNumber> {
+		pub votes_for: Vec<AccountId>,
+		pub votes_against: Vec<AccountId>,
+		pub status: ProposalStatus,
+		pub expiry: BlockNumber,
+	}
+
+	#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+	pub enum BridgeEvent {
+		FungibleTransfer(BridgeChainId, DepositNonce, ResourceId, U256, Vec<u8>),
+		NonFungibleTransfer(BridgeChainId, DepositNonce, ResourceId, Vec<u8>, Vec<u8>, Vec<u8>),
+		GenericTransfer(BridgeChainId, DepositNonce, ResourceId, Vec<u8>),
+	}
+
+	impl<A: PartialEq, B: PartialOrd + Default> ProposalVotes<A, B> {
+		/// Attempts to mark the proposal as approve or rejected.
+		/// Returns true if the status changes from active.
+		pub fn try_to_complete(&mut self, threshold: u32, total: u32) -> ProposalStatus {
+			if self.votes_for.len() >= threshold as usize {
+				self.status = ProposalStatus::Approved;
+				ProposalStatus::Approved
+			} else if total >= threshold && self.votes_against.len() as u32 + threshold > total {
+				self.status = ProposalStatus::Rejected;
+				ProposalStatus::Rejected
+			} else {
+				ProposalStatus::Initiated
+			}
+		}
+
+		/// Returns true if the proposal has been rejected or approved, otherwise false.
+		fn is_complete(&self) -> bool {
+			self.status != ProposalStatus::Initiated
+		}
+
+		/// Returns true if `who` has voted for or against the proposal
+		fn has_voted(&self, who: &A) -> bool {
+			self.votes_for.contains(who) || self.votes_against.contains(who)
+		}
+
+		/// Return true if the expiry time has been reached
+		fn is_expired(&self, now: B) -> bool {
+			self.expiry <= now
+		}
+	}
+
+	impl<AccountId, BlockNumber: Default> Default for ProposalVotes<AccountId, BlockNumber> {
+		fn default() -> Self {
+			Self {
+				votes_for: vec![],
+				votes_against: vec![],
+				status: ProposalStatus::Initiated,
+				expiry: BlockNumber::default(),
+			}
+		}
+	}
+
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		/// Origin used to administer the pallet
+		type BridgeCommitteeOrigin: EnsureOrigin<Self::Origin>;
+		/// Proposed dispatchable call
+		type Proposal: Parameter
+			+ Dispatchable<Origin = Self::Origin>
+			+ EncodeLike
+			+ GetDispatchInfo;
+		/// The identifier for this chain.
+		/// This must be unique and must not collide with existing IDs within a set of bridged
+		/// chains.
+		#[pallet::constant]
+		type BridgeChainId: Get<BridgeChainId>;
```

Comment:
the identical naming `BridgeChainId` might be misleading, it might be better to use `Get<u8>` , or rename to `Get<BridgechainIdType>`

Reply:
What I mean is the following codes:
```
pub type BridgeChainId = u8;
type BridgeChainId: Get<BridgeChainId>;
// later in runtime
type BridgeChainId = 1;
```
That's a bit misleading.
Despite not fully agreeing with it, let's not change it to avoid extra merge overhead later